### DISCUSSION TITLE
Open source WSL 1

### DISCUSSION THREAD

**Discussion by crramirez** [microsoft/WSL#5020] [General]
Hello,

WSL1 is still useful in most scenarios where WSL could be needed. Despite file system performance, file system incompatibilities, or incompatibilities very hard to solve, WSL1 could fit in most scenarios, and missing Kernel calls or bugs can be resolved one by one.

As I understand, the whole team and roadmap are focused on WSL2 to make it as convenient as WSL1. But as WSL2 has advantages that would be very hard to implement in WSL1. WSL1 has advantages like running in a process inside Windows, direct access to Windows partition, and does not require a Virtualization component that would be very hard to implement in WSL2. And simply, there are no resources to continue improving WSL1.

But, if the translation layer is open-sourced, the community can help in solving these issues and make WSL1 better.

Just a suggestion.

Regards,
Carlos

**Comment by WSLUser** [comment]
Technically this is dupe https://github.com/microsoft/WSL/issues/1 But now we have WSL2 and that issue is already closed. We do already have the console bits from the NT kernel open-source.

**Comment by bbulkow** [comment]
I need WSL1 for two reasons: file system performance, as I need to do all my work in NTFS because it's the shared file system ( one file system, multiple shells is the entire point ), and also the requirements to run WSL2 make it incompatible with VMware. If those two issues are resolved I am happy to switch to whatever is available.

That being said, I have a project which I can't run on NTFS because there is some subtle permission / softlink issue. So there are cases for both, if it wasn't for VMware I would have both running and choose each based on situation.
